jenkins/development.Dockerfile.j2

```diff
@@ -109,7 +130,9 @@ RUN \
         python3-email-validator \
         python3-flask \
         python3-flask-cors \
+{%- if fedora %}
         python3-flask-httpauth \
+{%- endif %}
```

Comment:
So, I just `diff`'d the original pair of `.Dockerfile`'s, and I think I understand the conditionalization here better now.  But, my initial instinct stands:  I think we have a set of requirements which this list is supposed to express, and it should be substantially the same on both/all platforms.  We'll still want conditionalization for cases where the names are different, and we'll want conditionalization for cases where the target package is unavailable; however, I think that the number of those cases is smaller than what we now have in this PR.  I think we're seeing some "extras" here, because the Jenkins `.Dockerfile` has evolved since the UBI one was cloned from it, and the UBI one hasn't kept up.

That said, we should ponder what we want the UBI build for.  It was originally intended for the Pbench-in-a-Can image, but Vishal has pointed out, rightly, that some of our images are ~huge~quite large.  _For that purpose_, we don't need the packages for the Agent, or testing, or other utility uses...and, it might be nice if we omitted them.  On the other hand, it might be nice to have a fully-functioning UBI image for other purposes...so, I don't really have any specific recommendations on this aspect.

In any case, I would like to see less conditionalization in this file if we can swing it.

Reply:
The goal is to have the Fedora development image have as few non-RPM installed package as possible ... ideally 0.

The UBI based image also needs the same, but we are not in that phase of productization to make a material impact on that.

I'd rather know what we have where now, rather than having to rediscover it later.

Further, we really want the development image requirements broken up a bit more:

 * common to agent and server
 * agent run-time RPMs
 * server run-time RPMs
 * composed agent & server run-time RPMs
 * agent & server test RPMs

That way, we can leverage those base images for our other containers.  But that is future work.  I'd rather keep the spirit of the previous two separate files than try to go further in this commit.

Reply:
I concur with all of that...I just want it to be clear why each of the conditionalized packages is conditionalized.

Taking `python3-bottle` for instance.  That package is required for the Agent's tool data sync.  So, the only reason that it should be conditionalized is if it is not available as an RPM.  As it happens, the RPM _is_ available on Fedora but (apparently) _not_ for the UBI.  So, I'm thinking that the conditionalization should be excluding UBI rather than selecting for Fedora.

That is, adopting the perspective that this recipe will be used for more than just two platforms (e.g., what about Centos; and what about RHEL 9 and Fedora Rawhide), I think we should exercise some care in how we construct the criteria.

As I said above, our requirements aren't dependent upon the platform (they are driven by the code), so at most we should be _excluding_ things here, not conditionally **_including_** things.

Reply:
Sure, I can invert all the conditionals to be `not ubi`.